pennylane/io.py

```diff
@@ -146,13 +156,99 @@ def circuit_loaded_qiskit_circuit():
+            the original SparsePauliOp acted on :math:`N` qubits, then this must be a sequence of
+            length :math:`N`
+
+    Returns:
+        Operator: The equivalent PennyLane operator.
+
+    .. note::
+
+        The wire ordering convention differs between PennyLane and Qiskit: PennyLane wires are
+        enumerated from left to right, while the Qiskit convention is to enumerate from right to
+        left. A ``SparsePauliOp`` term defined by the string ``"XYZ"`` applies ``Z`` on wire 0,
+        ``Y`` on wire 1, and ``X`` on wire 2.
+
+    **Example**
+
+    Consider the following script which creates a Qiskit ``SparsePauliOp``:
+
+    .. code-block:: python
+
+        from qiskit.quantum_info import SparsePauliOp
+
+        qiskit_op = SparsePauliOp(["II", "XY"])
+
+    The ``SparsePauliOp`` contains two terms and acts over two qubits:
+
+    >>> qiskit_op
+    SparsePauliOp(['II', 'XY'],
+                  coeffs=[1.+0.j, 1.+0.j])
+
+    To convert the ``SparsePauliOp`` into a PennyLane :class:`Operator`, use:
+
+    >>> import pennylane as qml
+    >>> qml.from_qiskit_op(qiskit_op)
+    I(0) + X(1) @ Y(0)
+
+    .. details::
+        :title: Usage Details
+
+        You can convert a parameterized ``SparsePauliOp`` into a PennyLane operator by assigning
+        literal values to each coefficient parameter. For example, the script
+
+        .. code-block:: python
+
+            import numpy as np
+            from qiskit.circuit import Parameter
+
+            a, b, c = [Parameter(var) for var in "abc"]
+            param_qiskit_op = SparsePauliOp(["II", "XZ", "YX"], coeffs=np.array([a, b, c]))
+
+        defines a ``SparsePauliOp`` with three coefficients (parameters):
+
+        >>> param_qiskit_op
+        SparsePauliOp(['II', 'XZ', 'YX'],
+              coeffs=[ParameterExpression(1.0*a), ParameterExpression(1.0*b),
+         ParameterExpression(1.0*c)])
+
+        The ``SparsePauliOp`` can be converted into a PennyLane operator by calling the conversion
+        function and specifying the value of each parameter using the ``params`` argument:
+
+        >>> qml.from_qiskit_op(param_qiskit_op, params={a: 2, b: 3, c: 4})
+        (
+            (2+0j) * I(0)
+          + (3+0j) * (X(1) @ Z(0))
+          + (4+0j) * (Y(1) @ X(0))
+        )
+
+        Similarly, a custom wire mapping can be applied to a ``SparsePauliOp`` as follows:
+
+        >>> wired_qiskit_op = SparsePauliOp("XYZ")
+        >>> wired_qiskit_op
+        SparsePauliOp(['XYZ'],
+              coeffs=[1.+0.j])
+        >>> qml.from_qiskit_op(wired_qiskit_op, wires=[3, 5, 7])
+        Y(5) @ Z(3) @ X(7)
+    """
+    try:
+        return load(qiskit_op, format="qiskit_op", params=params, wires=wires)
+    except ValueError as e:
+        if e.args[0].split(".")[0] == "Converter does not exist":
+            raise RuntimeError(_MISSING_QISKIT_PLUGIN_MESSAGE) from e
```

Comment:
Perhaps, but that would be inconsistent with `from_qiskit()` and we would still need to deal with dynamically importing the package and checking if that succeeds.

Reply:
I just have a minor preference to avoid entrypoints unless we need to use entrypoints. Entrypoints can cause confusion for devs (e.g., having to reinstall their editable PL version) and also for users on cloud notebooks (might have to restart the kernel multiple times).

Reply:
That is fair although I'm not sure using dynamic imports solves that problem (unless we force-reload the import every time). Do you think it's worth hosting a discussion with the broader team about these approaches?

Reply:
After discussing this offline, we agreed to leave the usage of entry points for now and revisit the matter later.